qa/test_template.sh

```diff
@@ -39,34 +38,36 @@ epilog=${epilog-:}
 # get the number of elements in `prolog` array
 numer_of_prolog_elms=${#prolog[@]}
 
-for i in `seq 0 $last_config_index`;
-do
+for i in `seq 0 $last_config_index`; do
     echo "Test run $i"
     # seq from 0 to number of elements in `prolog` array - 1
     for variant in $(seq 0 $((${numer_of_prolog_elms}-1))); do
         ${prolog[variant]}
         echo "Test variant run: $variant"
         # install packages
-        inst=$($topdir/qa/setup_packages.py -i $i -u $pip_packages --cuda ${CUDA_VERSION})
-        if [ -n "$inst" ]
-        then
-        pip install $inst -f /pip-packages
+        eval "packages=(`$topdir/qa/setup_packages.py -i $i -u $pip_packages --cuda ${CUDA_VERSION} | tr -d '[],'`)"
+        eval "packages_with_link=(`$topdir/qa/setup_packages.py -i $i -u $pip_packages --cuda ${CUDA_VERSION} --include-link | tr -d '[],'`)"
+        last_j=`expr ${#packages[@]} - 1`
+        for j in `seq 0 $last_j`; do
+            pkg="${packages[${j}]}"
+            pkg_with_link="${packages_with_link[${j}]}"
 
-        # If we just installed tensorflow, we need to reinstall DALI TF plugin
-        if [[ "$inst" == *tensorflow-gpu* ]]; then
-            pip uninstall -y nvidia-dali-tf-plugin || true
-            pip install /opt/dali/nvidia-dali-tf-plugin*.tar.gz
-        fi
-        fi
+            pip install "$pkg" -f /pip-packages --no-index || pip install "$pkg_with_link"
+
+            # If we just installed tensorflow, we need to reinstall DALI TF plugin
+            if [[ "$pkg" == *tensorflow-gpu* ]]; then
+                pip uninstall -y nvidia-dali-tf-plugin || true
+                pip install /opt/dali/nvidia-dali-tf-plugin*.tar.gz
+            fi
+        done
         # test code
         test_body
 
         # remove packages
-        remove=$($topdir/qa/setup_packages.py -r  -u $pip_packages --cuda ${CUDA_VERSION})
-        if [ -n "$remove" ]
-        then
-        pip uninstall -y $remove
-        fi
+        eval "remove=(`$topdir/qa/setup_packages.py -r  -u $pip_packages --cuda ${CUDA_VERSION} | tr -d '[],'`)"
+        for pkg in "${packages[@]}"; do
```

Comment:
Why loop?

Reply:
now the output of this script is an array, as opposed to a string